scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -140,16 +140,18 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   def entryPointAmmonite(): MultiSource = {
     require(input.isInstanceOf[Input.Ammonite])
     val builder = List.newBuilder[Source]
-    do {
+
+    builder += parseRuleAfterBOF(parseSourceImpl())
+    while (in.token match {
```

Comment:
There is no `do ... while()` construct for Scala 3

Reply:
since we have it in many places, and since in at least one you define a method, why not define our own:
```
// in companion object ScalametaParser, or perhaps even in scala.meta.common somewhere
def doWhile(body: => Unit)(cond: => Boolean): Unit = {
  body
  while (cond)
    body
}
```

Reply:
Good idea, done!